### ISSUE TITLE
[Discussion] Foxtrot team blocking deployment Wednesday-Thursday

### ISSUE THREAD

**Issue by amprokop** [department-of-veterans-affairs/caseflow#4180]
### Background

From ~3:00 pm Wednesday to ~1:00 pm Thursday, we didn’t have enough confidence to deploy the master branch to production. The Tango team wanted to deploy a hotfix to production (the specifics of that hotfix are out of scope here), but the Foxtrot team wasn’t confident in recent commits to master to allow a deploy. At around 330-4pm, it was discovered that there was a bug in the code currently on UAT, so we couldn’t deploy master until it was fixed or reverted. After a quick discussion, it was determined that we believed a fix could be achieved quickly, so we decided to roll forward. The fix was merged to master and deployed to UAT, but at ~630 pm, it was discovered that VBMS was down, so we delayed testing. The next morning, our dependencies were working, but we couldn’t successfully test all Certification scenarios to our satisfaction due to VACOLS/BGS/VBMS data not being set up. At ~12:00 pm Thursday, we reverted the commits that we weren’t sure about, and unblocked deploy.

### Five Whys

- We could not deploy from 3pm Wed to 1pm Thurs. Why?
- There was a bug on master. Why?
- We didn't fully validate code before merging to master. Why?
- There wasn't a clear/ideal test plan for this change. Why?
- There is friction for developers to test in lower envs when needed.

### Corrective Actions

- @nemodjuric already fixed the bug on master.
- Reviewers/devs should call out when changes need to be verified in UAT or connected to external dependencies (@amprokop/ tech leads)
- Understand/conquer test data setup issues (Devops team)
- Documentation should be adjusted to reflect that devs should explain the test plan in the PR (@lowellrex)
